build.xml

```diff
@@ -8,119 +8,268 @@
   <basename property="xspec.xml.base" file="${xspec.xml}" suffix=".xml" />
   <basename property="xspec.base" file="${xspec.xml.base}" suffix=".xspec" />
-
+  
   <property name="xspec.dir"
     value="${xspec.base.dir}/xspec" />
-
+  
   <property name="xspec.xsl"
     value="${xspec.dir}/${xspec.base}.xsl" />
   <property name="xspec.result.xml"
     value="${xspec.dir}/${xspec.base}-result.xml" />
   <property name="xspec.result.html"
     value="${xspec.dir}/${xspec.base}-result.html" />
   
-	<macrodef name="xspec"
-						description="Run XSpec tests">
-		<attribute name="xspec.xml" description="XSpec tests file" />
-		<sequential>
-			<local name="xspec.base.dir" />
-			<dirname property="xspec.base.dir" file="@{xspec.xml}" />
-			<local name="xspec.xml.base" />
-			<basename property="xspec.xml.base" file="@{xspec.xml}" suffix=".xml" />
-			<local name="xspec.base" />
-			<basename property="xspec.base" file="${xspec.xml.base}" suffix=".xspec" />
-			<antcall target="xspec.xspec" inheritall="false">
-				<param name="xspec.xml"
-							 location="@{xspec.xml}" />
-				<param name="xspec.base.dir"
-							 location="${xspec.base.dir}" />
-				<param name="xspec.dir"
-							 location="${xspec.base.dir}/xspec" />
-				<param name="xspec.base"
-							 value="${xspec.base}" />
-			</antcall>
-		</sequential>
-	</macrodef>
+  <!-- OXYGEN PATCH needed to create a custom Saxon, for example  -warnings:recover -strip:none -opt:10 -dtd:off -l:off -versionmsg:off -expand:on -outval:fatal -val:lax -->
+  <property name="saxon.custom.options" value=""/>
+  <!-- OXYGEN PATCH END -->
+  
+  <!-- OXYGEN PATCH needed to control the cleanup of all intermediate files -->
+  <property name="clean.output.dir" value="false"/>
+  <!-- OXYGEN PATCH END -->
+  
+  <!-- OXYGEN PATCH needed to customize the HTML result. -->
+  <property name="format.xspec.report" value="${xspec.project.dir}/src/reporter/format-xspec-report.xsl"/>
+  <!-- OXYGEN PATCH END -->
+  
+  <!-- OXYGEN PATCH needed to control the cleanup of all intermediate files -->  
+  <condition property="can.delete">
+    <and>
+      <equals arg1="${clean.output.dir}" arg2="true"/>
+    </and>
+  </condition>
+  <!-- OXYGEN PATCH END -->
+  
+  
+  
+  <!-- 
+    xmltask
+    download: http://www.oopsconsultancy.com/software/xmltask/
+    help: http://www.planetgeek.ch/2009/05/17/perform-xpath-queries-inside-ant/ 
+    
+  -->
+  <taskdef name="xmltask" classname="com.oopsconsultancy.xmltask.ant.XmlTask"/>
+  <taskdef resource="net/sf/antcontrib/antlib.xml"/>
+  
+  <property name="test.type" value="t"/>
+  <condition property="test.xslt">
+    <equals arg1="${test.type}" arg2="t"/>
+  </condition>
+  <condition property="test.schematron">
+    <equals arg1="${test.type}" arg2="s"/>
+  </condition>
+  
+  <!-- Copy of original XSpec file
+  will be overwritten if XSpec applies to Schematron -->
+  <property name="xspec.compiled.xml" location="${xspec.base.dir}/${xspec.base}-compiled-oxygen.xspec"/>
```

Comment:
Why does this file name have `-oxygen`?
It was puzzling to find such a file when I ran `build.xml` outside oXygen.

Reply:
I removed the `-oxygen` from the file name in https://github.com/xspec/xspec/commit/3b1a6f49032aef61941c06a5fd5bb6eeb82724d6#diff-2cccd7bf48b7a9cc113ff564acd802a8R113. This should be unnecessary once we fully integrate with  the oXygen ant build file.